drivers/usb/device/usb_dc_stm32.c

```diff
@@ -396,8 +396,9 @@ static int usb_dc_stm32_init(void)
 #endif /* CONFIG_USB_DEVICE_SOF */
 
 #if defined(CONFIG_SOC_SERIES_STM32H7X)
-	/* Currently assuming FS mode. Need to disable the ULPI clock on USB2.
-	 * Need to make this dependent on HS or FS config.
+	/* The USB2 controller only works in FS mode, but we still need to
+	 * disable its ULPI clock. As the USB1 controller is an HS one, this is
+	 * done in the common path.
```

Comment:
I'm still not getting why this step is required as this clock is off by default.
Is there a piece of code that activates it indirectly before this point?